include/nn/util/util_BitFlagSet.h

```diff
@@ -202,13 +208,20 @@ struct BitFlagSet {
 
     template <int BitIndex>
     struct Flag {
-        static const int Index = BitIndex;
-        // todo: figure out how to make this not break past 63
-        static const BitFlagSet Mask = 1 << Index;
+        static_assert(BitIndex < N);
+
+        static constexpr BitFlagSet buildMask() {
+            BitFlagSet tmp{{}};
```

Comment:
do we actually need the nested set of braces?

Reply:
I suppose one would zero initialize the array? would it perhaps be better to add braces to _storage member and then reduce it to one here?

Reply:
it should yeah (if unsure just check on godbolt :P)